src/Raven.Server/ServerWide/Commands/ModifyDatabaseWatchersCommand.cs

```diff
@@ -64,14 +65,22 @@ public override string UpdateDatabaseRecord(DatabaseRecord record, long etag)
             if (Watcher == null)
                 return null;
 
-            var newTaskKey = Watcher.GetTaskKey().ToString();
-            if (Watcher.CurrentTaskId != null && Watcher.CurrentTaskId != newTaskKey)
+            var newTaskId = (Watcher.Database + Watcher.Url).ToLower();
```

Comment:
Okay, make sense. In that case, just use the index number from raft when this was created.